Simple error crashes compiler -- missing () after predicate name in ensures clause

Using Dafny 2.1.1.10209. Mac OSX platform. The following program crashes the compiler. Stack trace follows code.

(And VS Code by the way doesn't handle stack dumps well.)


class CrashDafny
{
    predicate prop()
        ensures prop ==> true;  // Oops! Forgot () after prop
    {   true  }
} 

==== Stack Trace ====

[ERROR] FATAL UNHANDLED EXCEPTION: System.NullReferenceException: Object reference not set to an instance of an object
  at Microsoft.Dafny.Resolver.IsFunctionReturnValue (Microsoft.Dafny.Function fn, System.Collections.Generic.List`1[T] args, Microsoft.Dafny.Resolver+ResolveOpts opts) [0x00026] in <3b62ff0dffa34fe1b85139b4e6ef32a0>:0
  at Microsoft.Dafny.Resolver.ResolveExprDotCall (Microsoft.Boogie.IToken tok, Microsoft.Dafny.Expression receiver, Microsoft.Dafny.Type receiverTypeBound, Microsoft.Dafny.MemberDecl member, System.Collections.Generic.List`1[T] args, System.Collections.Generic.List`1[T] optTypeArguments, Microsoft.Dafny.Resolver+ResolveOpts opts, System.Boolean allowMethodCall) [0x00307] in <3b62ff0dffa34fe1b85139b4e6ef32a0>:0
  at Microsoft.Dafny.Resolver.ResolveNameSegment (Microsoft.Dafny.NameSegment expr, System.Boolean isLastNameSegment, System.Collections.Generic.List`1[T] args, Microsoft.Dafny.Resolver+ResolveOpts opts, System.Boolean allowMethodCall) [0x001cb] in <3b62ff0dffa34fe1b85139b4e6ef32a0>:0
  at Microsoft.Dafny.Resolver.ResolveExpression (Microsoft.Dafny.Expression expr, Microsoft.Dafny.Resolver+ResolveOpts opts) [0x007d1] in <3b62ff0dffa34fe1b85139b4e6ef32a0>:0
  at Microsoft.Dafny.Resolver.ResolveExpression (Microsoft.Dafny.Expression expr, Microsoft.Dafny.Resolver+ResolveOpts opts) [0x01879] in <3b62ff0dffa34fe1b85139b4e6ef32a0>:0
  at Microsoft.Dafny.Resolver.ResolveFunction (Microsoft.Dafny.Function f) [0x00285] in <3b62ff0dffa34fe1b85139b4e6ef32a0>:0
  at Microsoft.Dafny.Resolver.ResolveClassMemberBodies (Microsoft.Dafny.ClassDecl cl) [0x000c0] in <3b62ff0dffa34fe1b85139b4e6ef32a0>:0
  at Microsoft.Dafny.Resolver.ResolveTopLevelDecls_Core (System.Collections.Generic.List`1[T] declarations, Microsoft.Dafny.Graph`1[Node] datatypeDependencies, Microsoft.Dafny.Graph`1[Node] codatatypeDependencies) [0x00905] in <3b62ff0dffa34fe1b85139b4e6ef32a0>:0
  at Microsoft.Dafny.Resolver.ResolveModuleDefinition (Microsoft.Dafny.ModuleDefinition m, Microsoft.Dafny.ModuleSignature sig) [0x00271] in <3b62ff0dffa34fe1b85139b4e6ef32a0>:0
  at Microsoft.Dafny.Resolver.ResolveProgram (Microsoft.Dafny.Program prog) [0x004e0] in <3b62ff0dffa34fe1b85139b4e6ef32a0>:0
  at Microsoft.Dafny.Main.Resolve (Microsoft.Dafny.Program program, Microsoft.Dafny.ErrorReporter reporter) [0x00036] in <3b62ff0dffa34fe1b85139b4e6ef32a0>:0
  at Microsoft.Dafny.Main.ParseCheck (System.Collections.Generic.IList`1[T] files, System.String programName, Microsoft.Dafny.ErrorReporter reporter, Microsoft.Dafny.Program& program) [0x0001e] in <3b62ff0dffa34fe1b85139b4e6ef32a0>:0
  at Microsoft.Dafny.DafnyDriver.ProcessFiles (System.Collections.Generic.IList`1[T] dafnyFiles, System.Collections.ObjectModel.ReadOnlyCollection`1[T] otherFileNames, Microsoft.Dafny.ErrorReporter reporter, System.Boolean lookForSnapshots, System.String programId) [0x001d8] in <67f7d32a07d94740afb9e3c0dffff8fd>:0
  at Microsoft.Dafny.DafnyDriver.ThreadMain (System.String[] args) [0x0003d] in <67f7d32a07d94740afb9e3c0dffff8fd>:0
  at Microsoft.Dafny.DafnyDriver+<>c__DisplayClass1_0.<Main>b__0 () [0x00001] in <67f7d32a07d94740afb9e3c0dffff8fd>:0
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00017] in<829ce140006e4cad9124766ee7f51179>:0
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x0008d] in <829ce140006e4cad9124766ee7f51179>:0
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <829ce140006e4cad9124766ee7f51179>:0
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x00031] in <829ce140006e4cad9124766ee7f51179>:0
  at System.Threading.ThreadHelper.ThreadStart () [0x0000b] in <829ce140006e4cad9124766ee7f51179>:0

--- a/Source/Dafny/Resolver.cs
+++ b/Source/Dafny/Resolver.cs
@@ -13342,7 +13342,7 @@ namespace Microsoft.Dafny
       bool isFunctionReturnValue = true;
       // if the call is in post-condition and it is calling itself, and the arguments matches
       // formal parameter, then it denotes function return value.
-      if (opts.isPostCondition && opts.codeContext == fn) {
+      if (args != null && opts.isPostCondition && opts.codeContext == fn) {
         foreach (var arg in args) {
           if (arg is NameSegment) {
             var name = ((NameSegment)arg).Name;

--- a/Test/dafny4/git-issue182.dfy
+++ b/Test/dafny4/git-issue182.dfy
@@ -1,0 +1,8 @@
+// RUN: %dafny /compile:0 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+predicate method prop()
+ensures prop; // OOPS! Forgot () after prop
+{
+true
+}

--- a/Test/dafny4/git-issue182.dfy.expect
+++ b/Test/dafny4/git-issue182.dfy.expect
@@ -1,0 +1,2 @@
+git-issue182.dfy(5,8): Error: Postcondition must be a boolean (got () -> bool)
+1 resolution/type errors detected in git-issue182.dfy

